apps/web-client/src/lib/db/cr-sqlite/__tests__/reconciliation-order.test.ts

```diff
@@ -383,6 +377,522 @@ describe("Reconciliation order creation", () => {
+		expect(await getCustomerOrderLines(db, 2)).toEqual([expect.objectContaining({ isbn: "1", received: undefined })]);
+		expect(await getCustomerOrderLines(db, 3)).toEqual([expect.objectContaining({ isbn: "1", received: undefined })]);
+	});
+
+	it("reject customer order lines starting from last created customer order line (even if not part of the supplier order being reconciled)", async () => {
+		const db = await getRandomDb();
+
+		await upsertCustomer(db, { id: 1, displayId: "1" });
+		await upsertCustomer(db, { id: 2, displayId: "2" });
+		await upsertCustomer(db, { id: 3, displayId: "3" });
+
+		// First supplier order
+		await addBooksToCustomer(db, 1, ["1"]);
+		await addBooksToCustomer(db, 2, ["1"]);
+		// Second supplier order
+		await addBooksToCustomer(db, 3, ["1"]);
+		// Not placed (pending) - putting this here to make sure it doesn't affect rejection:
+		// even though last 'created', it's not placed yet - the customer 3's order should be rejected
+		await addBooksToCustomer(db, 4, ["1"]);
+
+		await createSupplierOrder(db, 1, [{ isbn: "1", quantity: 2, supplier_id: 1 }]);
+		await createSupplierOrder(db, 1, [{ isbn: "1", quantity: 1, supplier_id: 1 }]);
+
+		// NOTE: we're reconciling the 1st supplier order (orders are sorted in reverse chronological order)
+		const [, { id: supplierOrderId }] = await getPlacedSupplierOrders(db);
+
+		const reconOrderId = await createReconciliationOrder(db, [supplierOrderId]);
+		// Out of 2 books, only 1 arrived - 1 to reconcile (mark as received), 1 to reject (effectively marking as waiting to be re-ordered)
+		await addOrderLinesToReconciliationOrder(db, reconOrderId, [{ isbn: "1", quantity: 1 }]);
+		await finalizeReconciliationOrder(db, reconOrderId);
+
+		expect(await getCustomerOrderLines(db, 1)).toEqual([
+			expect.objectContaining({ isbn: "1", placed: expect.any(Date), received: expect.any(Date) })
+		]);
+		expect(await getCustomerOrderLines(db, 2)).toEqual([
+			expect.objectContaining({ isbn: "1", placed: expect.any(Date), received: undefined })
+		]);
+		// Even though customer order line 3 was not placed as part of the supplier order being reconciled, it was rejected by 1 book missing from the order
+		expect(await getCustomerOrderLines(db, 3)).toEqual([expect.objectContaining({ isbn: "1", placed: undefined, received: undefined })]);
+		// Not changed - control variable
+		expect(await getCustomerOrderLines(db, 4)).toEqual([expect.objectContaining({ isbn: "1", placed: undefined, received: undefined })]);
+	});
+
+	// NOTE: In a case where the book was overdelivered (considering the respective supplier order), but there are more in-progress customer orders
+	// for the same ISBN, it's ok for the overdelivered quantity to spill over to other customer order lines
+	it("reconcile more customer order lines than initially ordered (if such exist), even if not part of the current supplier order", async () => {
+		const db = await getRandomDb();
```

Comment:
isn't this the case Silvio mentioned in today's meeting would be doing too much?

Reply:
Yup, but this chunk of work is aging already, I thought to do this as one of the follow ups